.github/workflows/release.yml

```diff
@@ -60,31 +60,15 @@ jobs:
         npm version $INPUTS_INCREMENT --no-git-tag-version
         echo "new-version=$(npm pkg get version | tr -d '"')" >> $GITHUB_OUTPUT
 
-    # replace placeholders with current values in CHANGELOG.md
+    # replace placeholder with current values in CHANGELOG.md
+    # -E extended mode for \s support, -i inplace
+    # 0,: only operate on the first match.
     - name: Replace TBD entry in CHANGELOG.md
-      uses: actions/github-script@v7
       env:
-        NEW_VERSION: ${{ steps.increment-version.outputs.new-version }}
-      with:
-        script: |
-          const fsp = require('node:fs/promises');
-          const mdFile = 'CHANGELOG.md';
-
-          const newVersion = process.env.NEW_VERSION;
-          const currentDate = new Date().toISOString().replace(/T.*/i, '');
-
-          const mdContent = await fsp.readFile(mdFile, 'utf8');
-          const regex = /^##\s+\[Unreleased\]\s*$/im;
-          const newContent = mdContent.replace(regex, `## [${newVersion}] - ${currentDate}`);
-
-          if (mdContent !== newContent) {
-            await fsp.writeFile(mdFile, newContent);
-            if (process.env.INPUTS_DRY_RUN === 'true') {
-              core.info(newContent);
-            }
-          } else {
-            throw 'Pattern not found in CHANGELOG.md. Manually validate changelog.';
-          }
+        NEW_VERSION: ${{ steps.get-version.outputs.new-version }}
+      run: |
+        CURRENT_DATE=$(date +%Y-%m-%d)
+        sed -i -E "0,/^##\s+\[Unreleased\]\s*$/s//## [${NEW_VERSION}] - ${CURRENT_DATE}/" CHANGELOG.md
```

Comment:
```suggestion
        sed -i -E "0,/^##\s+\[Unreleased\].*$/s//## [${NEW_VERSION}] - ${CURRENT_DATE}/" CHANGELOG.md
```
the line we are targeting should _not_ have been modified manually. But I wouldn't be surprised if we found someone leaving a comment in this particular line at some point, i.e.

```md
## [Unreleased] (will be released after cds-dk x.z.y)
```

matching `.*` instead of `\s*` would catch this case.